File: music-infrastructure/src/main/java/com/music/sale/persistence/store/mapper/StorePersistenceMapper.java
package com.music.sale.persistence.store.mapper;

import com.music.sale.domain.store.Store;
import com.music.sale.persistence.store.entity.StoreEntity;
import org.springframework.stereotype.Component;

@Component
public class StorePersistenceMapper {

    public Store toDomain(StoreEntity entity) {
        return new Store(entity.getId());
    }

    public StoreEntity toEntity(Store store) {
        return new StoreEntity(
            store.getId(),
            "Default Store",
            "Default Description",
            "00000",
            "Default Address",
            null,
            null,
            null,
            "000-0000-0000",
            "000-00-00000",
            null,
            StoreEntity.StoreStatus.ACTIVE,
            1L
        );
    }
}

File: src/main/kotlin/com/music/sale/adapter/persistence/store/mapper/StorePersistenceMapper.kt
// Copyright (C) 2024 Your Name or Company
package com.music.sale.adapter.persistence.store.mapper

import com.music.sale.adapter.persistence.store.entity.StoreEntity
import com.music.sale.domain.store.model.Store
import org.springframework.stereotype.Component

@Component
class StorePersistenceMapper {
    fun toDomain(entity: StoreEntity): Store {
        return Store(
            id = entity.id,
        )
    }

    fun toEntity(store: Store): StoreEntity {
        return StoreEntity(
            id = store.id,
            name = "Default Store",
            description = "Default Description",
            zipcode = "00000",
            baseAddress = "Default Address",
            detailAddress = null,
            latitude = null,
            longitude = null,
            contactNumber = "000-0000-0000",
            businessNumber = "000-00-00000",
            imageUrl = null,
            status = StoreEntity.StoreStatus.ACTIVE,
            sellerId = 1L,
        )
    }
}
